### PR TITLE
Upgrade to match new monorepo build

### PR DESCRIPTION
**Pull request by noisekit**
- Ensure we use the exact version of dev libs. Because versions were added like `^0.0.0-123123` I noticed there was a total mixup with all the versions in yarn.lock, so updated the script accordingly
- `upgrade-synthetix:dev`
- Fix `error TS2532: Object is possibly 'undefined'`. That's something I haven't seen before. Not sure why this check failed after update.
- Switch to use correct library imports (officially exported values). No more deep-file imports from libs!

DEPLOYED to https://staking-git-noisekit-synthetixio.vercel.app and seems to be working ok

### CODE REVIEW

**Comment by bachstatter** [comment]
> Fix error TS2532: Object is possibly 'undefined'. That's something I haven't seen before. Not sure why this check failed after update.

This is due to this PR:
https://github.com/Synthetixio/js-monorepo/pull/263
It's merged but hasn't been released yet

**Comment by bachstatter** [comment]
> > It's merged but hasn't been released yet
> 
> I am using here the latest dev build that must include the change

Yea the dev build would reflect whatever is in the `dev` branch, so that make sense to me

File: package.json [hunk 1]
```diff
@@ -27,13 +27,13 @@
...
-    "@synthetixio/queries": "^3.0.5",
-    "@synthetixio/transaction-notifier": "^2.74.9",
-    "@synthetixio/wei": "^2.74.2",
+    "@synthetixio/contracts": "0.0.0-c5667822",
+    "@synthetixio/contracts-interface": "0.0.0-c5667822",
+    "@synthetixio/optimism-networks": "0.0.0-c5667822",
+    "@synthetixio/providers": "0.0.0-c5667822",
+    "@synthetixio/queries": "0.0.0-c5667822",
+    "@synthetixio/transaction-notifier": "0.0.0-c5667822",
+    "@synthetixio/wei": "0.0.0-c5667822",
```
Comment (bachstatter) [inline package.json:36]: I guess we wanna to a release and updates these before merging this PR?

**Review by 0xjocke — APPROVED** [review]
Everything looks good to me, except that we probably want a release on js-monorepo first?

File: package.json [hunk 2]
```diff
@@ -141,16 +140,24 @@
...
+    "@types/react": "^17.0.2",
     "@uniswap/sdk-core": "^3.0.0",
     "@uniswap/v3-core": "^1.0.0",
     "@uniswap/v3-periphery": "^1.0.0",
-    "ethers": "^5.6.4",
-    "sharp": "0.29.3",
     "browserslist": "^4.20.4",
     "caniuse-lite": "^1.0.30001355",
+    "ethers": "^5.6.4",
+    "sharp": "0.29.3",
```
Comment (bachstatter) [inline package.json:158]: Unrelated, but I'm pretty sure we can remove `sharp`

**Review by 0xjocke — APPROVED** [review]
Still looks good.

**Pull request merged**